datashuttle/utils/ssh.py

```diff
@@ -116,6 +117,7 @@ def verify_ssh_remote_host(remote_host_id: str, hostkeys: Path) -> bool:
     get the server key and present when connecting
     for manual validation.
     """
+    transport: paramiko.Transport
```

Comment:
why do you need to specify the type above if then you're using a `with` statement?

Reply:
Good question I kept getting a strange mypy error 
```
datashuttle\utils\ssh.py:138: error: Self? has no attribute "connect"  [attr-defined]
datashuttle\utils\ssh.py:139: error: Self? has no attribute "get_remote_server_key"  [attr-defined]
Found 2 errors in 1 file (checked 1 source file)
```
which I couldnt seem to solve or find much support online, I wonder if it is a bug related to paramiko's type hints. explicitly defining the type here solved the issue (though not sure why)